presto-hive/src/main/java/com/facebook/presto/hive/statistics/MetastoreHiveStatisticsProvider.java

```diff
@@ -212,6 +214,34 @@ private Estimate calculateDistinctValuesCount(Map<String, PartitionStatistics> s
                 DoubleStream::max);
     }
 
+    private Estimate calculateDataSize(Map<String, PartitionStatistics> statisticsByPartitionName, String column)
+    {
+        List<Double> knownPartitionDataSizes = statisticsByPartitionName.values().stream()
+                .map(stats -> {
+                    OptionalDouble averageColumnLength = stats.getColumnStatistics().get(column).getAverageColumnLength();
+                    OptionalLong rowCount = stats.getBasicStatistics().getRowCount();
+                    OptionalLong nullsCount = stats.getColumnStatistics().get(column).getNullsCount();
+                    if (!averageColumnLength.isPresent() || !rowCount.isPresent()) {
+                        return OptionalDouble.empty();
+                    }
+
+                    long nonNullsCount = rowCount.getAsLong() - nullsCount.orElse(0);
+                    return OptionalDouble.of(averageColumnLength.getAsDouble() * nonNullsCount);
+                })
+                .filter(OptionalDouble::isPresent)
+                .map(OptionalDouble::getAsDouble)
+                .collect(toImmutableList());
+
+        double knownPartitionDataSizesSum = knownPartitionDataSizes.stream().mapToDouble(a -> a).sum();
+        long partitionsWithStatsCount = knownPartitionDataSizes.size();
+        long allPartitionsCount = statisticsByPartitionName.size();
```

Comment:
This is not necessary correct. Statistics for some partitions might be missing. Pass a `hivePartitions.size()` as a `queriedPartitionsCount` instead.

Reply:
Good point.  I just copied this from the row count calculations, so it looks like we do that wrong too.  I'll add a fix.